src/c/realize_c_type.c

```diff
@@ -399,9 +401,9 @@ _realize_c_struct_or_union(builder_c_t *builder, int sindex)
                 assert(s->first_field_index >= 0);
                 ct = (CTypeDescrObject *)x;
                 ct->ct_size = (Py_ssize_t)s->size;
-                ct->ct_length = s->alignment;   /* may be -1 */
-                ct->ct_flags &= ~CT_IS_OPAQUE;
+                ct->ct_length = s->alignment; /* may be -1 */
                 ct->ct_flags_mut |= CT_LAZY_FIELD_LIST;
+                ct->ct_flags_mut &= ~CT_UNDER_CONSTRUCTION;
```

Comment:
I think this should remain as it was before:

1) We should do `ct->ct_flags &= ~CT_IS_OPAQUE` because `new_struct_or_union_type` unconditionally set `CT_IS_OPAQUE` and this isn't an opaque struct ( because of `!(s->flags & _CFFI_F_OPAQUE)`)

2) No need to do `ct->ct_flags_mut &= ~CT_UNDER_CONSTRUCTION` because `ct_flags_mut` was initialized to zero.

Even though we want `ct_flags` to be immutable, it's okay to clear `CT_IS_OPAQUE` here because we are effectively doing it as part of the struct construction; the construction is just kind of awkwardly split between `new_struct_or_union_type` and this function.